libethash-cl/CLMiner.cpp

```diff
@@ -377,10 +377,11 @@ void CLMiner::workLoop()
 
 			// Report hash count
 			addHashCount(m_globalWorkSize);
-		}
 
-		// Make sure the last buffer write has finished --
-		// it reads local variable.
+			// Make sure the last buffer write has finished --
+			// it reads local variable.
+			m_queue.finish();
```

Comment:
Sorry, a little late comment, but shouldn't this finish call be placed before `addHashCount`? Sounds like we want to add hash count only after queue is processed. I didn't investigate the code, just from the look of it.